src/backend/columnar/columnar_tableam.c

```diff
@@ -1589,13 +1592,46 @@ ColumnarProcessUtility(PlannedStmt *pstmt,
 								errmsg("only btree and hash indexes are supported on "
 									   "columnar tables ")));
 			}
+
+			/*
+			 * Since we don't support parallel scans on columnar tables, we
+			 * should dissuade postgres from choosing parallel scan for index
+			 * builds. Otherwise, CONCURRENT index builds would leave
+			 * an invalid index behind. Note that we want to do the same for
+			 * non-concurrrent index builds too. This is because, we want to
+			 * fall-back to sync scan instead of throwing an error.
+			 */
+			oldMaxParallelMaintenanceWorkers = max_parallel_maintenance_workers;
+			SetLocalMaxParallelMaintenanceWorkers(0);
 		}
 
 		RelationClose(rel);
 	}
 
 	PrevProcessUtilityHook(pstmt, queryString, context,
 						   params, queryEnv, dest, completionTag);
+
+	if (oldMaxParallelMaintenanceWorkers != -1)
+	{
+		/* altered by current command, roll it back */
+		SetLocalMaxParallelMaintenanceWorkers(oldMaxParallelMaintenanceWorkers);
```

Comment:
If we go the GUC route we should call set_config_option again with a NULL value (which is what RESET does).